lib/build.js

```diff
@@ -271,6 +272,122 @@ function getToken(self, job, pipeline) {
     return token;
 }
 
+/**
+ * @description Merges metadata from the parent event if available
+ * @param {number} parentEventId - The ID of the parent event
+ * @param {Object} mergedMeta - The current state of merged metadata
+ * @returns {Object} The updated merged metadata
+ */
+async function mergeParentEventMeta(parentEventId, mergedMeta) {
+    let resultMeta = mergedMeta;
+
+    if (parentEventId) {
+        // eslint-disable-next-line global-require
+        const EventFactory = require('./eventFactory');
+        const eventFactory = EventFactory.getInstance();
+        const parentEvent = await eventFactory.get(parentEventId);
+
+        resultMeta = _.merge(mergedMeta, parentEvent.meta);
+    }
+
+    return resultMeta;
+}
+
+/**
+ * @description Merges metadata from an external build
+ * @param {Object} mergedMeta - The current state of merged metadata
+ * @param {Object} parentJob - The job object associated with the parent build
+ * @param {Object} parentBuildMeta - The metadata from the parent build
+ * @returns {Object} The updated merged metadata
+ */
+function mergeExternalBuildMeta(mergedMeta, parentJob, parentBuildMeta) {
+    const jobName = parentJob.name;
+
+    mergedMeta.sd = mergedMeta.sd || {};
+    mergedMeta.sd[parentJob.pipelineId] = mergedMeta.sd[parentJob.pipelineId] || {};
+    mergedMeta.sd[parentJob.pipelineId][jobName] = _.merge(
+        mergedMeta.sd[parentJob.pipelineId][jobName] || {},
+        parentBuildMeta
+    );
+
+    return mergedMeta;
+}
+
+/**
+ * @description Merges metadata from parent builds if available
+ * @param {number|number[]} parentBuildId - The ID or list of IDs of parent builds
+ * @param {number} pipelineId - The ID of the pipeline
+ * @param {Object} mergedMeta - The current state of merged metadata
+ * @returns {Object} The updated merged metadata
+ */
+async function mergeParentBuildsMeta(parentBuildId, pipelineId, mergedMeta) {
+    // eslint-disable-next-line global-require
+    const BuildFactory = require('./buildFactory');
+    const buildFactory = BuildFactory.getInstance();
+    const parentBuildIds = Array.isArray(parentBuildId) ? parentBuildId : [Number(parentBuildId)];
+
+    const parentBuilds = await Promise.all(parentBuildIds.map(id => buildFactory.get(id)));
+
+    parentBuilds.sort((a, b) => a.endTime - b.endTime);
```

Comment:
Perhaps, `build.endTime` is string, so `a.endTime - b.endTime` is `NaN`.

Ref: https://github.com/screwdriver-cd/screwdriver/blob/9e9922e0dcf901fa0dd50747de244e889caf251a/plugins/builds/triggers/helpers.js#L683-L690

Reply:
Fixed.